src/server/service/kv.rs

```diff
@@ -1087,6 +1087,13 @@ fn extract_region_error<T>(res: &storage::Result<T>) -> Option<RegionError> {
             err.set_server_is_busy(server_is_busy_err);
             Some(err)
         }
+        Err(Error::Closed) => {
+            // TiKV is closing, return an RegionError to tell the client that this region is unavailable
+            // temporarily, the client should retry the request in other TiKVs.
+            let mut err = RegionError::new();
```

Comment:
It feels more like TiDB does not handle TCP close event properly. 🤔 What's the current behavior of TiDB when a TCP is closed?